src/line_shapes/get_line_shapes.jl

```diff
@@ -0,0 +1,141 @@
+
+"""
+Code to measure line bisectors and widths as a function of depth.
+Author: Alex Wise
+Created: September 2020
+"""
+
+""" get_line_shapes(line_list, clt, depths, v_width)
+For each spectrum in clt (chunklist timerseries), returns data frames for
+line bisector and width as a function of fractional line depth given in frac_depths
+note: frac_depths must be sorted from largest to smallest
+"""
+function get_line_shapes(line_list::A1, clt::AbstractChunkListTimeseries;
+                    frac_depths::A2 = 0.99:-0.01:0.05, v_width::Real=15e3,
+                    RV_estimate::Real=-5000.0) where { T1<:Real, A1<:AbstractArray{T1,1},
+                    T2<:Real, A2<:AbstractArray{T2,1}}
+    ll = copy(line_list)
+    n_lines = length(ll)
+    n_spectra = length(clt.chunk_list)
+    n_chunks = length(clt.chunk_list[1].data) # this assumes the number of chunks in every spectrum is the same
+    n_depths = length(frac_depths)
+    #allocate memory for line shapes
+    bisectors = zeros(n_lines,n_depths,n_spectra)
+    widths = zeros(n_lines,n_depths,n_spectra)
+    depths = zeros(n_lines,n_spectra)
+    #shift each entry in line list to RV estimate of star
+    ll .*= calc_doppler_factor(RV_estimate)
+    #find center of each chunk so that we can assign each line in line list to a chunk
+    chunk_centers = zeros(n_chunks)
+    for i in 1:n_chunks
+        chunk_centers[i] = (first(clt.chunk_list[1].data[i].λ) + last(clt.chunk_list[1].data[i].λ)) / 2.0
+    end
+    #pick which chunk to measure each line shape in
+    line_chunk_index = zeros(Int64,n_lines)
+    for i in 1:n_lines
+        line_chunk_index[i] = findmin(abs.((chunk_centers .- ll[i]) ./ chunk_centers))[2]
```

Comment:
What if one line shows up centered in two chunks, at different signal to noises?